### PR TITLE
fix(typescript_indexer): log TODO rather than error in indexer

### PR DESCRIPTION
**Pull request by ayazhafiz**
For errors that do not impede indexing of other entities in a TypeScript
source code, log a TODO rather than throwing an error to allow indexing
of the rest of the TS program to complete.

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar, @zzmp

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -508,7 +508,7 @@ class StandardIndexerContext implements IndexerHost {
     // TODO: update symbolNames table to account for context kind
 
     if (!declarations || declarations.length < 1) {
-      throw new Error('TODO: symbol has no declarations?');
+      console.warn(`TODO: symbol ${sym.name} has no declarations`);
```
Comment (zzmp) [inline kythe/typescript/indexer.ts]: Use the `todo` function here.

Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: unfortunately there is no `ts.Node` to pass to the todo function in this case because the only context is a symbol.

Comment (zzmp) [reply kythe/typescript/indexer.ts]: Could you add a comment explaining this for future readers?

Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: Done.

**Comment by ayazhafiz** [comment]
@evmar PTAL

**Comment by ayazhafiz** [comment]
@evmar PTAL

**Comment by evmar** [comment]
Still relevant?

**Comment by ayazhafiz** [comment]
no

**Pull request closed without merging**